shielder/cli/src/main.rs

```diff
@@ -177,6 +180,48 @@ async fn do_deposit(
     }
 }
 
+async fn do_merge(
+    contract: Shielder,
+    connection: Connection,
+    cmd: MergeCmd,
+    app_state: &mut AppState,
+) -> Result<()> {
+    let MergeCmd {
+        first_deposit_id,
+        second_deposit_id,
+        caller_seed,
+        proving_key_file,
+        ..
+    } = cmd;
+
+    let seed = match caller_seed {
+        Some(seed) => seed,
+        None => Password::new("Seed of the merging account (the tokens owner):")
+            .without_confirmation()
+            .prompt()?,
+    };
+    let connection = SignedConnection::from_connection(connection, keypair_from_string(&seed));
+
+    let first_deposit = app_state
+        .get_deposit_by_id(first_deposit_id)
+        .ok_or(anyhow!("Cannot match first deposit id to actual deposit!"))?;
+    let second_deposit = app_state
+        .get_deposit_by_id(second_deposit_id)
+        .ok_or(anyhow!("Cannot match second deposit id to actual deposit!"))?;
+
```

Comment:
optionally, I would put here at least 2 basic checks:
 - `first_deposit != second_deposit`
 - their token_ids are identical

we could handle such errors much better than waiting for the contract failure + we avoid calling contract

Reply:
Totally reasonable, I have added the requisite checks.